### PR TITLE
fix: Remove user field from tracker preheat [DHIS2-18001]

### PR DESCRIPTION
**Pull request by enricocolasante**
Remove user fields from `TrackerPreheat`.
`NotesConverterService` needs `TrackerUserService` to correctly create a `Note` object. This is a temporary solution that will be fixed in upcoming PRs.

**NEXT**
- Continue harmonization on tracker importer. At the moment, we have `User`, `UserDetails` and `UserInfoSnapshot` used in different places.
- Harmonize exporters

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterService.java [hunk 1]
```diff
@@ -201,10 +202,10 @@ private Event from(
       result.setUid(!StringUtils.isEmpty(event.getEvent()) ? event.getEvent() : event.getUid());
       result.setCreated(now);
       result.setStoredBy(event.getStoredBy());
-      result.setCreatedByUserInfo(preheat.getUserInfo());
+      result.setCreatedByUserInfo(UserInfoSnapshot.from(CurrentUserUtil.getCurrentUserDetails()));
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterService.java:205]: I am not sure how some of these classes are currently tested. Changes like this make them less easy to unit test. Not sure what the strategy is but if we want to unit test such transformations we should pass the user into the function.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/AbstractTrackerPersister.java [hunk 2]
```diff
@@ -377,7 +378,10 @@ private void delete(
             : entityManager.merge(trackedEntityAttributeValue));
 
     logTrackedEntityAttributeValueHistory(
-        preheat.getUsername(), trackedEntityAttributeValue, trackedEntity, ChangeLogType.DELETE);
+        CurrentUserUtil.getCurrentUsername(),
```
Comment (muilpp) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/AbstractTrackerPersister.java:381]: I guess it's just a matter of preference, but why import this method statically?

**Review by muilpp — APPROVED** [review]

**Review by teleivo — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterServiceTest.java [hunk 3]
```diff
@@ -80,7 +82,10 @@ class EventTrackerConverterServiceTest extends TestBase {
 
   private static final Date today = new Date();
 
-  private final NotesConverterService notesConverterService = new NotesConverterService();
+  @Mock public TrackerUserService trackerUserService;
```
Comment (ameenhere) [inline dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterServiceTest.java:85]: I thought Mocking was banned from today 😆

**Review by ameenhere — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterService.java (hunk 1 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/converter/EventTrackerConverterService.java:205]: Maybe we can discuss this and agree on the best way to use the user and have testable code.
At the moment, we need to inject the user in the tests.

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/AbstractTrackerPersister.java (hunk 2 above)
Comment (enricocolasante) [reply dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/AbstractTrackerPersister.java:381]: Yeah, I can add static imports

**Pull request merged**